sphinx/writers/latex.py

```diff
@@ -2206,13 +2210,13 @@ def depart_literal_strong(self, node):
     def visit_abbreviation(self, node):
         # type: (nodes.Node) -> None
         abbr = node.astext()
-        self.body.append(r'\sphinxstyleabbreviation{')
-        # spell out the explanation once
-        if node.hasattr('explanation') and abbr not in self.handled_abbrs:
-            self.context.append('} (%s)' % self.encode(node['explanation']))
-            self.handled_abbrs.add(abbr)
+        if node.hasattr('explanation'):
+            self.body.append(r'\gls{')
+            self.abbr.add('\\newacronym{' + abbr + '}{' + abbr + '}{' +
+                    node['explanation'] + '}\n')
```

Comment:
in case input has duplicated possibly diverging explanations, the original code stored only the first explanation ; here it seems ``\newacronym` will be inserted in LaTeX multiple times in such cases. This does not cause problem?